src/expr/src/vector_op/agg/approx_count_distinct.rs

```diff
@@ -122,17 +121,25 @@ impl Aggregator for ApproxCountDistinct {
         self.return_type.clone()
     }
 
-    fn update_with_row(&mut self, input: &DataChunk, row_id: usize) -> Result<()> {
+    fn update_single(&mut self, input: &DataChunk, row_id: usize) -> Result<()> {
         let array = input.column_at(self.input_col_idx).array_ref();
         let datum_ref = array.value_at(row_id);
         self.add_datum(datum_ref);
-
         Ok(())
     }
 
-    fn update(&mut self, input: &DataChunk) -> Result<()> {
+    fn update_multi(
+        &mut self,
+        input: &DataChunk,
+        start_row_id: usize,
+        end_row_id: usize,
+    ) -> Result<()> {
         let array = input.column_at(self.input_col_idx).array_ref();
-        for datum_ref in array.iter() {
+        for datum_ref in array
+            .iter()
+            .skip(start_row_id)
+            .take(end_row_id - start_row_id)
```

Comment:
.take will stop when we reach the end of the array? Should assert start_row_id and end_row_id in range.

Reply:
I think the cases where `start_row_id` and `end_row_id` are mis-specified to be out of input data chunk is rare. More possible is to call `update_multi` with overlapped ranges, which can't be detected in `update_multi` impls. Adding asserts to `start/end_row_id` here doesn't help a lot and makes code ugly.🤔

Reply:
Yes, I'd agree on that. I'm just thinking we could add asserts to make things easier if there's really a bug in the future. Maybe `debug_assert!` which will only take effect in debug mode. Anyway, it looks good to me overall if we already ensure everything is in range before calling this function.